File: dairy/public/js/pricing_rule.js
frappe.ui.form.on(
    'Pricing Rule', {
    
    //     frm.fields_dict.applicable_for.new_doc = function(){
    
    set_options_for_applicable_for: function(frm){
        if(frm.doc.selling == '1') {
        
            frm.set_df_property('applicable_for','options',[" ", "Customer", "Customer Group", "Territory", "Sales Partner", "Campaign", "Address"]);
            
        }
        if(frm.doc.buying== '1'){
            frm.set_df_property('applicable_for','options',[" ", "Supplier", "Supplier Type"]);

        }
        frm.refresh_field('applicable_for');
    },  

});

File: dairy/public/py/pricing_rule.py
# from __future__ import unicode_literals
# import frappe
# import json
# import copy
# from frappe import throw, _
# from frappe.utils import flt, cint, getdate

# from frappe.model.document import Document

# from six import string_types

# @frappe.whitelist()
# def apply_pricing_rule(args, doc=None):
# 	"""
# 		args = {
# 			"items": [{"doctype": "", "name": "", "item_code": "", "brand": "", "item_group": ""}, ...],
# 			"customer": "something",
# 			"customer_group": "something",
# 			"territory": "something",
# 			"supplier": "something",
# 			"supplier_group": "something",
#             "address": "something",
# 			"currency": "something",
# 			"conversion_rate": "something",
# 			"price_list": "something",
# 			"plc_conversion_rate": "something",
# 			"company": "something",
# 			"transaction_date": "something",
# 			"campaign": "something",
# 			"sales_partner": "something",
# 			"ignore_pricing_rule": "something"
# 		}
# 	"""
#     if